### PR TITLE
[SPARK-15533][SQL] Deprecate Dataset.explode

### PR DESCRIPTION
**Pull request by sameeragarwal**
## What changes were proposed in this pull request?

This patch deprecates `Dataset.explode` and documents appropriate workarounds to use `flatMap()` or `functions.explode()` instead.
## How was this patch tested?

N/A

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 1]
```diff
@@ -1556,7 +1556,7 @@ class Dataset[T] private[sql](
   }
 
   /**
-   * :: Experimental ::
+   * :: Deprecated ::
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: remove this we don't have a label like this

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 2]
```diff
@@ -1576,10 +1576,23 @@ class Dataset[T] private[sql](
    *   val bookCountPerWord = allWords.groupBy("word").agg(countDistinct("title"))
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: we can remove the old examples now

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 3]
```diff
@@ -1596,7 +1609,7 @@ class Dataset[T] private[sql](
   }
 
   /**
-   * :: Experimental ::
+   * :: Deprecated ::
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: remove this

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 4]
```diff
@@ -1605,10 +1618,23 @@ class Dataset[T] private[sql](
    *   ds.explode("words", "word") {words: String => words.split(" ")}
    * }}}
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: remove old example

**Comment by sameeragarwal** [comment]
Removed labels and converted the examples to use the alternatives.

**Comment by rxin** [comment]
LGTM pending tests.

**Comment by rxin** [comment]
Merging in master/2.0.

**Pull request closed without merging**